src/fidesops/schemas/dataset.py

```diff
@@ -107,6 +107,8 @@ class FidesopsMeta(BaseModel):
     """Optionally specify the allowable field length. Fidesops will not generate values that exceed this size."""
     return_all_elements: Optional[bool]
     """Optionally specify to query for the entire array if the array is an entrypoint into the node. Default is False."""
+    read_only: Optional[bool]
+    """Optionally specify if a field is read-only, meaning it can't be updated or deleted."""
```

Comment:
The approach we've been taking for masking requests is to take a row from the access request and mask select fields according to the rule targets. These masked fields are then be posted via the update body. When following this approach for Stripe, it was necessary to include other required fields as part of the payload, even if they weren't being masked. However, if we took the original row, merged it with the masked fields, and posted the entire row back to the API, we would receive errors for certain fields that were not allowed in update requests.

I made a previous suggestion based on our legacy implementation to be able to flag fields with "update" or "delete" flags based on which actions were supported for that field. This would have meant having to go back and adding update/delete flags for SaaS integrations that we've already completed (which didn't seem like the right approach). The approach I settled on instead was to flag fields as "read-only" which lets us omit those fields from being included when generating the body for an update request. I am open to other suggestions if this approach does not seem flexible enough or of this flag should not be part of the more general fidesops_meta.